PWGJE/FlavourJetTasks/AliAnalysisTaskHFSubstructure.cxx

```diff
@@ -999,10 +994,10 @@ Bool_t AliAnalysisTaskHFSubstructure::FillHistograms()
 	
 	}
       }
-      delete Matched_Truth_Particle;
-      delete D_Candidate;
+      // delete Matched_Truth_Particle;
+      // delete D_Candidate;
     }
-    delete Track_Container;
```

Comment:
From what I see you take the track container from the AliAnalysisTaskEmcal via GetParticleContainer. The AliAnalysisTaskEmcal however keeps ownership over the particle containers (including track containers). This should result in a double delete, and you possibly get into trouble the next time you access the track container.